### PR TITLE
DT-2948 Add endpoint to get all active users, returning user detail i…

### PR DESCRIPTION
**Pull request by karenmillermoj**
…ncluding email

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/resource/UserResource.kt [hunk 1]
```diff
@@ -163,6 +163,20 @@ class UserResource(
     ) @RequestParam @NotEmpty lastName: String
   ): List<UserSummaryWithEmail> = userService.findUsersByFirstAndLastNames(firstName, lastName)
 
+  @PreAuthorize("hasRole('ROLE_MANAGE_NOMIS_USER_ACCOUNT')")
+  @GetMapping("/all")
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/resource/UserResource.kt:167]: Not sure whether the naming of the endpoint should be **/users/active** ?

Comment (neilmendum) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/resource/UserResource.kt:167]: `users/active` sounds reasonable

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/resource/UserResourceIntTest.kt [hunk 2]
```diff
@@ -434,6 +434,86 @@ class UserResourceIntTest : IntegrationTestBase() {
...
+        .headers(setAuthorisation(roles = listOf("ROLE_MANAGE_NOMIS_USER_ACCOUNT")))
+        .exchange()
+        .expectStatus().isOk
+        .expectBody()
+        .jsonPath("$").value<JSONArray> {
+          assertThat(it.map { m -> (m as Map<*, *>)["username"] })
+            .contains("fred.smith")
+            .contains("frederica.jones")
+            .contains("marco.rossi")
+            .hasSize(3)
```
Comment (neilmendum) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/resource/UserResourceIntTest.kt:512]: You could make one of the users with `createUsers` inactive to test that it isn't included in the results

Comment (karenmillermoj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/resource/UserResourceIntTest.kt:512]: Good idea

**Review by neilmendum — APPROVED** [review]
LGTM

**Pull request merged**